src/apply.c

```diff
@@ -650,7 +649,12 @@ int git_apply_to_tree(
 	 */
 	for (i = 0; i < git_diff_num_deltas(diff); i++) {
 		delta = git_diff_get_delta(diff, i);
-
+		if (delta->status == GIT_DELTA_ADDED &&
```

Comment:
We probably need to handle the rename case as well, when the `new_file.path` exists in the postimage already.